aws/data.yaml

```diff
@@ -2,30 +2,32 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: data
+  name: data-dev
   labels:
     environment: dev
     app: data
 spec:
   selector:
     matchLabels:
+      environment: dev
       app: data
   replicas: 1
   template:
     metadata:
       labels:
+        environment: dev
         app: data
     spec:
       containers:
         - name: data
-          image: docker.io/healthmapidha/dataservice:latest
+          image: docker.io/healthmapidha/dataservice:0.0.3
           ports:
             - containerPort: 3000
           env:
             - name: DB_CONNECTION_STRING
               valueFrom:
                 secretKeyRef:
-                  name: data-878ckgc42b
+                  name: data-dev-ddhhd4mmhf
```

Comment:
Silly question perhaps, but why is this binding name machine generated? Most examples I see have a human readable binding "category" for name, and the key is the specific secret being used.

Reply:
I'm using kubernetes kustomize to generate the secrets, you can check out the official docs here
https://kubernetes.io/docs/concepts/configuration/secret/
The machine generated suffix is here for that reason:
_Note: When a Secret is generated, the Secret name is created by hashing the Secret data and appending this value to the name. This ensures that a new Secret is generated each time the data is modified._